### PR TITLE
Use explicit inputs for languages

### LINKED ISSUE #27: Use explicit inputs
### What happened?

Using explicit inputs instead of implicitly inherited env vars leaves us safe from issues such as [this one](https://github.com/pulumi/ci-mgmt/issues/961).

### Example

See https://github.com/pulumi/pulumi-civo/actions/runs/9395511963/job/25875398234

### Output of `pulumi about`

n/a

### Additional context

_No response_

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

### PR DESCRIPTION
**Pull request by guineveresaenger**
Fixes #27.

Testing this via https://github.com/pulumi/pulumi-civo/pull/515.

### CODE REVIEW

**Review by iwahbe — APPROVED** [review]
This is a breaking change to pulumi-package-publisher, so we should probably call it out in the release step.

**Comment by guineveresaenger** [comment]
This failed on p-civo because the java action did not have access to the composite Action's Inputs. Trying again.

File: action.yml [hunk 1]
```diff
@@ -76,14 +88,22 @@ runs:
     - name: Nodejs - Publish to NPM
       uses: ./.pulumi-package-publish/lang/nodejs
       if: (contains(inputs.sdk, 'nodejs') || (contains(inputs.sdk, 'all') && !contains(inputs.sdk, '!all'))) && !contains(inputs.sdk, '!nodejs')
+      env:
+        NODE_VERSTION: ${{ inputs.node-version }}
```
Comment (iwahbe) [inline action.yml:92]: Can we pass this via `with` instead of `env`?

Comment (guineveresaenger) [reply action.yml:92]: I did not find any evidence to this effect - the example given didn't show using `with` on this level: https://docs.github.com/en/actions/creating-actions/creating-a-composite-action

I wouldn't be surprised if it does work but given how long our Actions are taking to kick off today, I'm going to leave this as a refinement. :/

**Pull request merged**